### PR TITLE
Better CSS

### PR DESCRIPTION
**Pull request by davepagurek**
- Makes the job types page not be a too-wide table
- Styles all forms (in event receivers and in general create/edit forms) so that labels are all aligned
- Use styled buttons everywhere
- Fix back links that go to the wrong pages
- Changed "api_key" to "API Key" in the UI
- Changed "job type" to "job template" in the UI, I figured that's maybe more understandable? thoughts?
- Fixed a bug in the React code where state was getting mutated before `setState` finishes, which messed up diffing. Now it uses `cloneDeep` and assigns to a new object when copying state so that these early mutations don't happen.

### CODE REVIEW

File: app/javascript/packs/receiver.jsx [hunk 1]
```diff
@@ -167,12 +171,17 @@ export default class Receiver extends React.Component {
...
       }, callback);
     }
   }
 
+  actionSavedState(index) {
+    return this.props.event_actions[index];
+  }
+
   removeAction(actionIndex) {
+    console.log(actionIndex);
```
Comment (AndrewMcBurney) [inline app/javascript/packs/receiver.jsx]: Could you remove this `console.log` please?

File: app/views/job_types/_form.html.slim [hunk 2]
```diff
@@ -1,4 +1,4 @@
-= form_for @job_type do |f|
+= form_for @job_type, html: {class: 'form'} do |f|
```
Comment (AndrewMcBurney) [inline app/views/job_types/_form.html.slim]: Could you add spaces in between the hash braces, please? 

```slim
= form_for @job_type, html: { class: 'form' } do |f|
```

File: app/views/job_types/index.html.slim [hunk 3]
```diff
@@ -1,37 +1,34 @@
...
+            pre = job_type.environment_variables
+          .field
+            label Timeout
+            span = job_type.timeout
+        .column.actions
+          = link_to "Edit", edit_job_type_path(job_type), class: "button button--secondary"
+          = button_to "Destroy", job_type, method: :delete, class: "button button--secondary"
           - if current_user.workers.active.any?
-            = button_to 'Spawn', { controller: "jobs", action: "create", job: { job_type_id: job_type.id } }, method: :post
+            = button_to 'Spawn', { controller: "jobs", action: "create", job: { job_type_id: job_type.id } }, method: :post, class: "button button--secondary"
```
Comment (AndrewMcBurney) [inline app/views/job_types/index.html.slim]: Could you please make the `'Spawn'` use double quotes?

File: app/views/job_types/index.html.slim [hunk 4]
```diff
@@ -1,37 +1,34 @@
...
+            button.button.button--secondary(disabled=true) Wait for one of your workers to come online to spawn this job.
           - else
-            button(disabled=true) Create a worker to spawn this job
+            button.button.button--secondary(disabled=true) Create a worker to spawn this job
 
 
 br
 
-= link_to 'New Job type', new_job_type_path
+= link_to 'New Job type', new_job_type_path, class: "button button--primary"
```
Comment (AndrewMcBurney) [inline app/views/job_types/index.html.slim]: Could you please make the `'New Job type'` use double quotes?

File: app/views/workers/index.html.slim [hunk 5]
```diff
@@ -21,8 +21,8 @@ table
           = api_key.name
         td
           = api_key.key
-        td = link_to 'Edit', edit_api_key_path(api_key)
-        td = button_to 'Destroy', api_key, method: :delete
+        td = link_to 'Edit', edit_api_key_path(api_key), class: "button button--secondary"
```
Comment (AndrewMcBurney) [inline app/views/workers/index.html.slim]: Could you please change the single quotes to double quotes here? Thanks homie ❤️

**Comment by davepagurek** [comment]
Changed!

**Pull request merged**